src/reactapp/src/components/shippingAddress/components/ShippingAddressFormikProvider.jsx

```diff
@@ -136,6 +150,18 @@ function ShippingAddressFormikProvider({ children }) {
     forceViewMode,
   ]);
 
+  // whenever country value changed, we will find the country entry from the countryList
+  // so that we can toggle the validation on the `region` field
+  useEffect(() => {
+    if (countryList && countryValue) {
+      const regionRequired = !!countryList.find(
+        (country) => country.id === countryValue
+      )?.state_required;
+
+      toggleRegionRequiredSchema(regionRequired);
+    }
+  }, [countryValue, countryList]);
+
```

Comment:
This logic is repeating in the billing and shipping section. so we need to utilize this commonly. May be move logic to existing common hooks or create a new hook here would be better

Reply:
This will be handled in a future pull request